### ISSUE TITLE
Title/image special handling shouldn't be on by default (or at least should be documented and have a way to disable)

### ISSUE THREAD

**Issue by kanitw** [vega/vega-tooltip#694]
The following blocks in  https://github.com/vega/vega-tooltip/blob/next/src/formatValue.ts#L19-25 seems pretty problematic for me

```ts
    if (title) {
      content += `<h2>${valueToHtml(title)}</h2>`;
    }

    if (image) {
      content += `<img src="${valueToHtml(image)}">`;
    }
```

doesn't seem to be documented and may cause unintended effect.

For example, if there is a field in the data source called `title`, it will automatically becomes "title" in the output even though users may not intend to do so. 

<img width="433" alt="image" src="https://user-images.githubusercontent.com/111269/188762871-97c7b8ca-b7f0-46d5-bab5-af84b75c212a.png">

[spec](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289AEEABNCTY46nILaZjMNgHcassobvqIcHA-OGks5YYINIJqhjg+FIJWshCGAGZs6obESJpsDLFQqRCUiiqmKKigDOqCaPmYSAQIbMQ0alIy8gC+SpjqfhAJ6ghoANqgWYJQDIKmcOXKpgwIlAAqdCF5SBDlOphLzQC6SsjqANZooJgAnuHlXk6YeZhsbII6OEcgUHKYcLLX6FOVIM2tIA+r2UTjIzwAHs84vVBMpygBVADKhgA4uo2BBVm0zhN0ABHBh+damGikP5KE5QmFw9AAdUSsMcqlR6MxNxx5QJRLoJLJ-2aQA) --> I need to add calculate to simulate that there is a field title in the example. 

I don't think we should do this by default since it will surprise people more.  

cc: @domoritz @nyurik  (You might have an idea why we had this block.)

**Comment by domoritz** [comment]
I agree we should document the default handling of special fields and add an option to disable it. When we added it, it made sense as a quick way to have nicer tooltips.

**Comment by kanitw** [comment]
I agree it's a nice feature to have, but I think it's not a good default since it's too magical that these two fields are treated inconsistently.

**Comment by domoritz** [comment]
I don't feel strongly about it but I do like it as a default since it is hard for people to know how to configure the tooltips from a spec alone (many don't have direct access to Vega Tooltip in their applications).

**Comment by kanitw** [comment]
> many don't have direct access to Vega Tooltip in their applications

That's fair.   

I think I'm ok with if the key is more esoteric like `vlTooltipTitle` or `__vlTooltipTitle__` or something. (same for image). 

There are data tables with a field `title` everywhere (like movies title), so having this title by default seems pretty problematic.

**Comment by kanitw** [comment]
I think what we can do is to introduce `titleKey` and `imageKey` config and default to our preferred format that's hard to collide (e.g., `tooltip:title`).  

For existing applications that already rely on "title" and "image" as a trick can set titleKey to "title" if they prefer.

**Comment by domoritz** [comment]
Agreed. For imageKey, it should be a set imageKeys since there can be more than one image.

**Comment by beanscg** [comment]
Bean Labs did a read-only pass on the tooltip formatter. `src/formatValue.ts` destructures `title` and `image` out of the value object and special-renders them before the remaining fields; `src/defaults.ts` exposes `formatTooltip`, `sanitize`, and `baseURL`, but I did not see a default option for renaming or disabling those special keys.

That suggests two small acceptance paths: document the reserved-key behavior clearly, or add an explicit `titleKey`/`imageKey` style option with tests covering collision with ordinary data fields.

No PR opened from this pass.
